state/upgrades.go

```diff
@@ -815,3 +815,64 @@ func addStorageInstanceConstraints(st *State) error {
 	}
 	return nil
 }
+
+var splitBatchRemoveSize = 1000
+
+// SplitLogCollections moves log entries from the old single log collection
+// to the log collection per environment.
+func SplitLogCollections(st *State) error {
+	session := st.MongoSession()
+	db := session.DB(logsDB)
+	oldLogs := db.C("logs")
```

Comment:
This collection won't necessarily be there once we drop it. I guess we drop it below so even though we create it here, it will be gone later. But if there's an error in the upgrade step, and logs had already been dropped before, we will now leave it behind when we shouldn't